Sources/MapboxNavigation/CPRouteChoice.swift

```diff
@@ -0,0 +1,34 @@
+import CarPlay
+import MapboxDirections
+
+@available(iOS 12.0, *)
+extension CPRouteChoice {
+    
+    struct RouteResponseUserInfo {
+        
+        static let key = "com.mapbox.navigation.cpRouteChoice.routeResponse"
+        
+        /**
+         Route response from the Mapbox Directions service.
+         */
+        let response: RouteResponse
+        
+        /**
+         Index of the route.
+         */
+        let routeIndex: Int
+        
+        /**
+         Options, which were used for calculating results from the Mapbox Directions service.
+         */
+        let options: DirectionsOptions
+    }
+    
+    var routeResponseFromUserInfo: RouteResponseUserInfo? {
```

Comment:
Is it important to put this data into a formal struct, or should it remain a loosely typed dictionary with defined keys for forward-compatibility?

Reply:
I do not think that it's critical to store values in separate keys of the dictionary. Storing data in internal struct also helps with readability.